sfputil/main.py

```diff
@@ -1314,9 +1312,12 @@ def update_firmware_info_to_state_db(port_name):
         state_db = SonicV2Connector(use_unix_socket_path=False, namespace=namespace)
         if state_db is not None:
             state_db.connect(state_db.STATE_DB)
-            active_firmware, inactive_firmware = platform_chassis.get_sfp(physical_port).get_transceiver_info_firmware_versions()
-            state_db.set(state_db.STATE_DB, 'TRANSCEIVER_INFO|{}'.format(port_name), "active_firmware", active_firmware)
-            state_db.set(state_db.STATE_DB, 'TRANSCEIVER_INFO|{}'.format(port_name), "inactive_firmware", inactive_firmware)
+            transceiver_firmware_info_dict = platform_chassis.get_sfp(physical_port).get_transceiver_info_firmware_versions()
+            if transceiver_firmware_info_dict is not None:
+                active_firmware = transceiver_firmware_info_dict.get('active_firmware', 'N/A')
+                inactive_firmware = transceiver_firmware_info_dict.get('inactive_firmware', 'N/A')
+                state_db.set(state_db.STATE_DB, 'TRANSCEIVER_FIRMWARE_INFO|{}'.format(port_name), "active_firmware", active_firmware)
+                state_db.set(state_db.STATE_DB, 'TRANSCEIVER_FIRMWARE_INFO|{}'.format(port_name), "inactive_firmware", inactive_firmware)
```

Comment:
@mihirpat1 do we need to update here now that Xcvrd dom thread already updates?

Reply:
@prgeor - This is not a must have anymore.
However, I added it to improve latency for the update in TRANSCEIVER_FIRMWARE_INFO table since the DomInfoUpdateTask can take more than a minute to update the firmware information.
Also, this function will be called when DOM polling is disabled for the port. Hence, this will ensure that TRANSCEIVER_FIRMWARE_INFO table is not simultaneously updated by DomInfoUpdateTask and sfputil.

Reply:
@prgeor - I have removed this now.

Reply:
@prgeor - I have added this back as per our discussion.